.github/workflows/sonar.yaml

```diff
@@ -0,0 +1,36 @@
+name: Build
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    types: [opened, synchronize, reopened]
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
+      - name: Set up JDK 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+      - name: Cache SonarCloud packages
+        uses: actions/cache@v1
+        with:
+          path: ~/.sonar/cache
+          key: ${{ runner.os }}-sonar
```

Comment:
Is this correct? It will never be recached once done? In another repo, I see
```
key: ${{ runner.os }}-sonar-${{ steps.date.outputs.date }}
```
which looks more reasonable. (However, I don't understand `restore-keys`, so I may be missing something.)

Reply:
That's the default with the issue of potentially having a very large cache over the long term. For now, I think we can just proceed with the default.